openwisp_controller/config/models.py

```diff
@@ -166,8 +170,45 @@ class Meta(AbstractTemplate.Meta):
 
     def clean(self):
         self._validate_org_relation('vpn')
+        if self.sharing == 'import':
+            self._get_remote_template()
         super(Template, self).clean()
 
+    def _get_remote_template(self):
+        try:
+            with urllib.request.urlopen(self.url) as response:
```

Comment:
why aren't you using the `requests` library? Aren't we already using it in this module?

Reply:
This is what i intend to use to get the external template from the URL. 
The code gets the Template and set the various fields so that `clean` method doesn't hold the process for validation errors. Equally, in case the external template have a `vpn-client`, the code ensures that the various VPN requirements, gotten from external source are created before setting the `self.vpn` so that validation doesn't fail.